[docs] Bash in Gitpod (#385)
gitpod-io/website: src/docs/bash_in_gitpod.md

# Bash
Bash is the default shell in Gitpod. But if you're developing Bash scripts yourself, you may want to configure additional language support for Bash.

> ## Workspace Config

Reviewer: No good Bash example repo?

I think https://github.com/inlets/inlets has some Shell scripts, but maybe not enough to be a relevant example here.

Actually here maybe we could create a small demo repo just for Bash? E.g. you import the code of "99 Bottles of Beer" from http://www.99-bottles-of-beer.net/language-bash-98.html and have a `.gitpod.yml` that runs it.

Author: Done

> ### VSCode Extensions

Reviewer: I think this title level is not useful.

```suggestion
```

#### ShellCheck

To use ShellCheck in Gitpod, first, we must install the `shellcheck` tool. Here is a Dockerfile to get you started.
```Dockerfile
FROM gitpod/workspace-full

USER gitpod

> RUN brew update && brew install shellcheck

Reviewer: No need to run `brew update` here.

```suggestion
RUN brew install shellcheck
```

```

> You should also install the shellcheck VS Code extension in Gitpod, e.g. by adding a `.gitpod.yml` configuration file to your repository that looks like this (notice the `vscode` section):

Reviewer: Suggestion to clarify a bit more:

```suggestion
You should also install the shellcheck VS Code extension in Gitpod, e.g. by adding a `.gitpod.yml` configuration file to your repository that looks like this (notice the `vscode` extensions section):
```

# Bash

> Bash is pre-installed as the default shell in the main Gitpod workspace-image, however, there isn't much support for shell scripting out-of-the-box.

Reviewer: In general, we should avoid negative language about Gitpod, e.g. "Gitpod doesn't have great support for X".

We've decided to support the most common languages out-of-the-box where it makes sense, but Dockerfiles and VS Code extensions allow anyone to add support for almost any language relatively easily.

Here is a suggestion:
```suggestion
Bash is the default shell in Gitpod. But if you're developing Bash scripts yourself, you may want to configure additional language support for Bash.
```

## Workspace Config

### VSCode Extensions

> #### Linting

Reviewer: I think there is no need to have a very deep title-hierarchy for language-specific tutorials.

Ideally only one or two levels of titles should suffice. Here `Workspace Config` > `VSCode Extensions` > `Linting` > `ShellCheck` is a bit too deep (I don't think we're planning to list many more VS Code extensions for Bash, and even if we did, we could have independent sections for each without a huge hierarchy).

##### ShellCheck

> To use ShellCheck in Gitpod, first, we must install it. Here is a Dockerfile to get you started.

Reviewer: "it" is confusing here (the VS Code extension or the CLI?)

```suggestion
To use ShellCheck in Gitpod, first, we must install the `shellcheck` tool. Here is a Dockerfile to get you started.
```

> ```dockerfile

Reviewer: I think the correct language name is `Dockerfile` and not `dockerfile`. But that's not very important so feel free to dismiss this suggestion.

```suggestion
```Dockerfile
```

FROM gitpod/workspace-full

USER gitpod

RUN sudo apt-get -q update && \
    sudo apt-get install -yq shellcheck && \

>     sudo rm -rf /var/lib/apt/lists/*

Reviewer: Please check if this can be install with `brew install shellcheck`, and if so replace the complicated `apt-get` snippet.

In general, using `brew` in our example Dockerfiles can make them much shorter and simpler (but first we should double-check that using `brew` actually works for that).

```

> Here is a .gitpod.yml with the shellcheck VSCode Extension already installed.

Reviewer: I agree that there are already other docs pages that explain what a `.gitpod.yml` is and how you can install VS Code extensions.

But I think it wouldn't be nice here to assume that every reader is already an expert in Gitpod configurations.

Please find a way to make this text easier to understand for brand-new Gitpod users, e.g.:

```suggestion
You should also install the shellcheck VS Code extension in Gitpod, e.g. by adding a `.gitpod.yml` configuration file to your repository that looks like this (notice the `vscode` section):
```

I think we should use more beginner-friendly language every time we mention a `.gitpod.yml` or (especially) `.gitpod.Dockerfile` in the docs.

Also, it never hurts to add a link to the relevant docs page for more explanation (I personally _always_ make [.gitpod.Dockerfile](https://www.gitpod.io/docs/42_config_docker/) a link to https://www.gitpod.io/docs/42_config_docker/, so that any confused users can click the link and learn more about what this thing is we're talking about).
